compiler/rustc_codegen_llvm/src/allocator.rs

```diff
@@ -82,4 +88,41 @@ pub(crate) unsafe fn codegen(tcx: TyCtxt<'_>, mods: &mut ModuleLlvm, kind: Alloc
         }
         llvm::LLVMDisposeBuilder(llbuilder);
     }
+
+    // rust alloc error handler
```

Comment:
Is there any reason you duplicated this code rather than adding a new entry to [`ALLOCATOR_METHODS`](https://github.com/rust-lang/rust/blob/85fbf49ce0e2274d0acf798f6e703747674feec3/compiler/rustc_ast/src/expand/allocator.rs#L32)?

Reply:
@bjorn3 
1. `ALLOCATOR_METHODS` is also used [elsewhere](https://github.com/rust-lang/rust/blob/85fbf49ce0e2274d0acf798f6e703747674feec3/compiler/rustc_builtin_macros/src/global_allocator.rs#L37-L47) to build other stuff, which I didn't want to clobber
2. the alloc error handler has a different function signature
3. the alloc error handler has a different `name`

Reply:
Ah, ok.

Reply:
@bjorn3 feel free to refactor the stuff without side effects